core/src/main/java/org/dozer/config/resolvers/YAMLSettingsResolver.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2005-2017 Dozer Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dozer.config.resolvers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import org.apache.commons.io.FilenameUtils;
+import org.dozer.config.Settings;
+import org.dozer.util.DozerClassLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Settings resolver which loads up a yaml file, such as dozer.yaml
+ */
+public class YAMLSettingsResolver implements SettingsResolver {
+
+    private static final Logger LOG = LoggerFactory.getLogger(YAMLSettingsResolver.class);
+
+    private static final String ROOT_PARENT = "dozer";
+    private static final String CACHE_PARENT = "cache";
+    private static final String EL_PARENT = "el";
+    private static final String CLASSES_PARENT = "classes";
+
+    private static final String ENABLED_PROPERTY = "enabled";
+    private static final String CONVERTER_BY_DEST_TYPE_CACHE_MAX_SIZE_PROPERTY = "converter_by_dest_type_maxsize";
+    private static final String SUPER_TYPE_CHECK_CACHE_MAX_SIZE_PROPERTY = "super_type_maxsize";
+    private static final String CLASS_LOADER_BEAN_PROPERTY = "classloader";
+    private static final String PROXY_RESOLVER_BEAN_PROPERTY = "proxy_resolver";
+
+    private final SettingsResolver fallthrough;
+    private final DozerClassLoader classLoader;
+    private final String configFile;
+    private Map<String, Object> properties = new HashMap<>();
+
+    public YAMLSettingsResolver(SettingsResolver fallthrough, DozerClassLoader classLoader, String configFile) {
+        this.fallthrough = fallthrough;
+        this.classLoader = classLoader;
+        this.configFile = configFile;
+    }
+
+    @Override
+    public void init() {
```

Comment:
I would suggest to split this method by responsibilities, i.e. locating/reading the file and processing map from yaml.